### ISSUE TITLE
Flutter project is not launching from Xcode

### ISSUE THREAD

**Issue by gaaclarke** [flutter/flutter#126689]
## steps to reproduce
1) start a new project
1) paste in the repo code from https://github.com/flutter/flutter/issues/126620
1) Add in the http dependency (**I think this is key**)
1) launch it from the command-line to an iPhone (notice that it executes fine)
1) Open up ios/Runner/Runner.xcworkspace in Xcode
1) Hit the play button on xcode

I'm unable to reproduce it with just the standard flutter counter app.

## error log
```
2023-05-12 09:32:15.934100-0700 Runner[15751:6468463] Metal API Validation Enabled
2023-05-12 09:32:16.558641-0700 Runner[15751:6468463] [VERBOSE-2:dart_vm_data.cc(20)] VM snapshot invalid and could not be inferred from settings.
2023-05-12 09:32:16.558875-0700 Runner[15751:6468463] [VERBOSE-2:dart_vm.cc(268)] Could not set up VM data to bootstrap the VM from.
2023-05-12 09:32:16.558991-0700 Runner[15751:6468463] [VERBOSE-2:dart_vm_lifecycle.cc(85)] Could not create Dart VM instance.
2023-05-12 09:32:16.559088-0700 Runner[15751:6468463] [VERBOSE-3:shell.cc(139)] Check failed: vm. Must be able to initialize the VM.
dyld4 config: DYLD_LIBRARY_PATH=/usr/lib/system/introspection DYLD_INSERT_LIBRARIES=/Developer/usr/lib/libBacktraceRecording.dylib:/Developer/usr/lib/libMainThreadChecker.dylib:/Developer/Library/PrivateFrameworks/DTDDISupport.framework/libViewDebuggerSupport.dylib
```

## doctor
```
aaclarke-macbookpro3:to_image_data_bug aaclarke$ flutter doctor -v
[✓] Flutter (Channel main, 3.11.0-5.0.pre.22, on macOS 13.3.1 22E772610a darwin-arm64, locale en)
    • Flutter version 3.11.0-5.0.pre.22 on channel main at /Users/aaclarke/dev/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision cab29b2034 (3 days ago), 2023-05-09 18:38:49 +0200
    • Engine revision 824cd09b8c
    • Dart version 3.1.0 (build 3.1.0-87.0.dev)
    • DevTools version 2.23.1
```

**Comment by leighajarett** [comment]
Does it still run when using `flutter run` @gaaclarke ?

**Comment by gaaclarke** [comment]
> Does it still run when using `flutter run` @gaaclarke ?

Yep, works fine in that case.  It only fails when launching from xcode, which is required for debugging the engine.  I'm not sure if there is a use case beyond that.

**Comment by stuartmorgan** [comment]
Are you running with a local engine here? The invalid snapshot sounds suspiciously like the kind of error that building with one engine and running with another can generate.

**Comment by gaaclarke** [comment]
I can't remember the exact steps so I just went back and did:
```sh
$ flutter create to_image_data_bug
$ cd to_image_data_bug
$ flutter pub add http
# paste in the main.dart
$ flutter run
$ open ios/Runner.xcworkspace
# press play
```

And it worked with a flutter from 3 hours ago.  While investigating this bug I would have been using a local engine to try to debug it, so that appears to be required to reproduce the bug.  From memory I believe I got into a state where using flutter without a local engine was displaying the error.

Typically my workflow is:
1) From the command line specify the local engine (ex `flutter --local-engine=ios_debug_unopt --local-engine-src-path=/Users/aaclarke/dev/engine/src/ run`)
1) Open the `.xcworkspace` and hit run to debug the engine.

I can't play around with it more right now since my GOMA broke today and I need to fix it.

> The invalid snapshot sounds suspiciously like the kind of error that building with one engine and running with another can generate.

Hitting play on xcode should rebuild things.  Do you think the xcode build flow is not smart enough to rebuild in all cases and usually just works by accident?  I just happened to run into a case that broke compatibility between my local engine and flutter's engine?  That would make sense, but it would be nice if our iOS build could detect that case.
